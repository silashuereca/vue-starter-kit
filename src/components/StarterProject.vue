<template>
  <div>
    <p v-text="message" />
    <p v-text="doubleValue" />
  </div>
</template>

<script lang="ts">
import { defineComponent, computed } from "vue";
import { createPinia, setActivePinia } from "pinia";
import { useCounterStore } from "../stores/counter";
export default defineComponent({
  setup() {
    const pinia = createPinia();
    setActivePinia(pinia); //if I don't set this then the component test will fail (this needs further investigation as I shouldn't have to do this here since the instance has already been initiated in main.ts)
    const store: any = useCounterStore();
    const message: string = "Vue Starter Project";
    setInterval(() => {
      store.increment();
    }, 1000);
    return {
      message,
      doubleValue: computed(() => store.doubleCount),
    };
  },
});
</script>

<style scoped>
p {
  font-size: 40px;
}
</style>
