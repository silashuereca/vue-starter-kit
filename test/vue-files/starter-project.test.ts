import { mount } from "@vue/test-utils";
import StarterProject from "../../src/components/StarterProject.vue";

test("make sure elements are rendered correctly", () => {
  const wrapper = mount(StarterProject);

  expect(wrapper.element).toMatchSnapshot();
});
