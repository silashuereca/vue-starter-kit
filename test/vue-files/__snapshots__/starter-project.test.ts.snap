// Vitest Snapshot v1

exports[`make sure elements are rendered correctly 1`] = `
<div
  data-v-b9c8eaea=""
>
  <p
    data-v-b9c8eaea=""
  >
    Vue Starter Project
  </p>
  <p
    data-v-b9c8eaea=""
  >
    0
  </p>
</div>
`;
